### DISCUSSION TITLE
plugin for s3-compatible object storage

### DISCUSSION THREAD

**Discussion by kenob** [dokku/dokku#5304] [Show and tell]
I prototyped a plugin for s3-compatible object storage, backed by [minio](https://docs.min.io/docs/minio-client-complete-guide). It's been working fine for me on my local server, but I'm willing to harden it + add docs and tests if there is interest from the community.

Repository: https://github.com/kenob/dokku-objectstore

**Comment by azlekov** [comment]
Hey, nice one! These days I was looking for such plugin. I will give it a try, but few questions 

How can I access or expose the console?
How the bucket creation is managed on service start, how can I create a default bucket?

**Reply by azlekov** [reply]
About exposing the console what I tried:

1. Exposing the service via the command and then map the port following 
2. This guide https://pro.dokku.com/docs/operating/exposing-publicly/

**Reply by josegonzalez** [reply]
You could use this to expose a plugin: https://github.com/dokku/service-proxy

**Reply by kenob** [reply]
For bucket creation, that doesn't happen at creation right now, but there is some code [here](https://github.com/kenob/dokku-objectstore/blob/d8ad1bd8b524b76e1a6b8848e5fb7b8d620d6717/functions#L82) that runs some commands against the server, so that could create a bucket. 

However, this feels like a thing that should be done by the client, akin to creating a table in Postgres maybe.
